plugins/lsp/conform-nvim.nix

```diff
@@ -0,0 +1,111 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.conform-nvim;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.conform-nvim =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "conform-nvim";
+
+      package = helpers.mkPackageOption "conform-nvim" pkgs.vimPlugins.conform-nvim;
+
+      # Map of filetype to formatters
+      formattersByFt =
+        helpers.defaultNullOpts.mkNullable (types.attrsOf types.anything)
+        "see documentation"
+        ''
+          # Map of filetype to formatters
+          formattersByFt =
+            {
+              lua = [ "stylua" ];
+              # Conform will run multiple formatters sequentially
+              python = [ "isort" "black" ];
+              # Use a sub-list to run only the first available formatter
+              javascript = [ [ "prettierd" "prettier" ] ];
+              # Use the "*" filetype to run formatters on all filetypes.
+              "*" = [ "codespell" ];
+              # Use the "_" filetype to run formatters on filetypes that don't
+              # have other formatters configured.
+              "_" = [ "trim_whitespace" ];
+             };
+        '';
+
+      formatOnSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+            timeoutMs =
+              mkOption
```

Comment:
you should also user helpers.defaultNullOpts.mkBool there (or provide a default value) else a user must provide both even if he only sets one